### PR TITLE
Removing patch dependencies scope restriction to release branch only

### PR DESCRIPTION
**Pull request by PaurushGarg**
*Issue #, if available:*

*Description of changes:*

- [x]  Includes maven cache in prbuild to make subsequent pr builds faster.
- [x]  Updates patch workflows and scripts so that they are not scoped to only release branches.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/main-build.yml [hunk 1]
```diff
@@ -339,7 +337,7 @@ jobs:
         with:
           path: |
             ~/.m2/repository/io/opentelemetry/
-          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/**/opentelemetry-java-*.patch') }}
+          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java-*.patch') }}
```
Comment (bryan-aguilar) [inline .github/workflows/main-build.yml:340]: Is this key correct? This will not hash `.github/patches/opentelemetry-java.patch`

Comment (PaurushGarg) [reply .github/workflows/main-build.yml]: Yes, thanks for highlighting this. I have now updated to `.github/patches/opentelemetry-java*.patch`

File: .github/workflows/main-build.yml [hunk 2]
```diff
@@ -70,7 +70,7 @@ jobs:
         with:
           path: |
             ~/.m2/repository/io/opentelemetry/
-          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java-*.patch') }}
+          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java*.patch') }}
```
Comment (bryan-aguilar) [inline .github/workflows/main-build.yml:73]: Did you do a code search and verify that you changed all instances of this cache key?

Comment (PaurushGarg) [reply .github/workflows/main-build.yml:73]: yes I changed in both pr and main build. I don't see this hash key anywhere else in the code - did the search.

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**